lib/util/uint8array_utils.js

```diff
@@ -131,11 +131,49 @@ shaka.util.Uint8ArrayUtils.equal = function(arr1, arr2) {
 
 
 /**
- * Convert a Uint8Array to a string which can be used as a key in a dictionary.
+ * Generates a random unique number to use as a key for a given Unit8Array
+ * or returns the key for a a given Unit8Array
  * @param {!Uint8Array} arr
  * @return {string}
  */
-shaka.util.Uint8ArrayUtils.key = function(arr) {
-  return Array.prototype.join.apply(arr);
-};
-
+shaka.util.Uint8ArrayUtils.key = (function() {
+
+  var cache = {};
```

Comment:
Ah yes I see, we didn't consider that scenario. What do you think is best here, pull in an external MD5/SHA1 library (with an appropriate license), or come up with something custom? We had mentioned the possibility of using WebCrypto to do the hashing; this would work, but this is already one of the features that requires a secure origin, so until EME itself requires secure origin I guess this isn't feasible?